src/Umbraco.Web.UI.Client/src/views/propertyeditors/multiurlpicker/multiurlpicker.controller.js

```diff
@@ -88,14 +90,16 @@ function multiUrlPickerController($scope, angularHelper, localizationService, en
                             link.isMedia = model.target.isMedia;
                         }
 
-                        link.name = model.target.name || model.target.url || model.target.anchor;
```

Comment:
I don't think there's a need to set the name to the URL or anchor, as that's already shown below the name.